### ISSUE TITLE
training on whole posts

### ISSUE THREAD

**Issue by ysmiraak** [argsim/argsim#1]
using the internet argument corpus https://nlds.soe.ucsc.edu/iac

- extract the posts from the json files
- adapt current model (we may have to reduce the model size) for training
- decide whether to use sentencepiece sampling

**Comment by ysmiraak** [comment]
how many posts in the datasets are under length `t` as sentence pieces

|    t |    iac | reason |
|------|--------|--------|
|  256 | 0.7170 | 0.8438 |
|  512 | 0.8865 | 0.9553 |
| 1024 | 0.9644 | 0.9905 |
| 2048 | 0.9924 | 0.9963 |

**Comment by ysmiraak** [comment]
currently the maximum length is set to 512, longer posts are truncated (see `encode_capped` in `util_sp.py`)

training 2 models without sentencepiece sampling
- 256-512 dimensional, batch size 128 (~1.7 it/s)
- 512-1024 dimensional, batch size 64 (~1.4 it/s)

both for 11 rounds (10k steps per round), roughly 36 and 18 epochs respectively.  let's see which option is better, and then we'll try training with sampled inputs.

**Comment by ysmiraak** [comment]
halving the dimensions turns out to be extremely detrimental.  even though it could go through twice amount of data, it performs significantly worse than the model with doubled width after the same number of training steps.

the large model with halved batch size is currently the master model.  i will keep training it for 11 rounds at a time, til convergence.

another model under training right now uses sentencepiece sampling (alpha=0.5), with (usually) different sampled sequences as source and target.

**Comment by ysmiraak** [comment]
training on whole posts takes much longer to converge than training on sentences.
the original learning rate decay was likely too extreme.

here are the basics.
each round is `1e4` steps, and it's inverse is the accelerate `1e-4`.
so let `r = accelerate * step`, this `r` is the basis for the training schedule:
- word dropout rate `sigmoid r`
- kl loss weight `tanh r`
- learning rate decay

originally the learning rate was `lr / (r + 1)` with `lr` being the initial learning rate.
currently it's `lr / ((sqrt r) + 1)`.
so the learning rate starts out as `1e-3` and gracefully decrease to `2e-4` at round 10 and to `1e-4` at round 33.

the new learning rate decay is comparable to the schedule used by the transformer.
after the warmup stage, their learning rate is `1 / (sqrt (step * dim))` with `dim = 512` being the model dimension, which is roughly half of our new learning rate after 20 rounds.

**Comment by ysmiraak** [comment]
done
